internal/backend/azure/token_credential.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+// clientAssertionCredential authenticates an application with assertions provided by a callback function.
+type clientAssertionCredential struct {
+	assertion, file string
+	client          confidential.Client
+	lastRead        time.Time
+}
+
+// clientAssertionCredentialOptions contains optional parameters for ClientAssertionCredential.
+type clientAssertionCredentialOptions struct {
+	azcore.ClientOptions
+}
+
+// newClientAssertionCredential constructs a clientAssertionCredential. Pass nil for options to accept defaults.
+func newClientAssertionCredential(tenantID, clientID, authorityHost, file string, options *clientAssertionCredentialOptions) (*clientAssertionCredential, error) {
```

Comment:
Is there some documentation which describes the necessary ceremony to get the credentials? Just from looking at the code, I can't tell whether it makes sense or not.

Reply:
Sorry not except the microsoft example here: v

Reply:
@ekarlso I do NOT appreciate copying large parts of an example without proper attribution as required by the license.